File: me/seeking/value/Value.java
package me.seeking.value;



import java.util.function.Supplier;

public class Value<V> {
    private String name;
    private V value;
    private final Supplier<Boolean> visitable;

    public Value(String name, V value, Supplier<Boolean> visitable) {
        this.name = name;
        this.visitable = visitable;
        this.value = value;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public V getValue() {
        return this.value;
    }

    public void setValue(V value) {
        this.value = value;
    }

    public boolean isVisitable() {
        return visitable.get();
    }
}

File: src/main/java/me/seeking/value/Value.kt
package me.seeking.value

import java.util.function.Supplier

/**
 * This file is a part Seeking
 */

open class Value<V>(name: String, value: V, visitable: Supplier<Boolean>) {
    private lateinit var name: String
    var value: V
    private val visitable: Supplier<Boolean>

    init {
        this.name = name
        this.visitable = visitable
        this.value = value
    }


    fun isVisitable(): Boolean {
        return visitable.get()
    }
}

